### PR TITLE
[Java] step-4: projecting to kafka

### PR DESCRIPTION
**Pull request by octonato**

### CODE REVIEW

File: docs-source/docs/modules/microservices-tutorial/examples/shopping-analytics-service-java/src/main/java/shopping/analytics/ShoppingCartEventConsumer.java [hunk 1]
```diff
@@ -42,13 +42,13 @@ static void init(ActorSystem<?> system) {
...
-        RestartSource // <3>
+        RestartSource // <1>
                 .onFailuresWithBackoff(minBackoff, maxBackoff, randomFactor, () -> {
                     return Consumer
-                            .committableSource(consumerSettings, Subscriptions.topics(topic)) // <1>
+                            .committableSource(consumerSettings, Subscriptions.topics(topic)) // <2>
                             .mapAsync(1, msg ->
                                     handleRecord(msg.record()).thenApply(done -> msg.committableOffset()))
-                            .via(Committer.flow(committerSettings)); // <2>
+                            .via(Committer.flow(committerSettings)); // <3>
```
Comment (octonato) [inline docs-source/docs/modules/microservices-tutorial/examples/shopping-analytics-service-java/src/main/java/shopping/analytics/ShoppingCartEventConsumer.java:51]: changed ordering to follow the reading order

File: docs-source/docs/modules/microservices-tutorial/pages/projection-grpc-client.adoc [hunk 2]
```diff
@@ -143,7 +143,7 @@ include::example$shopping-cart-service-scala/src/main/scala/shopping/cart/SendOr
 
 In the order we need to include the list of items and their quantities. That information is not included in the `ShoppingCart.CheckedOut` event, but we can retrieve it by asking the `ShoppingCart` entity for it.
 
-If the call to the `ShoppingOrderService` fails the returned [.group-scala]#`Future[Done]`# [.group-java]#`CompletionStage[Done]`# is failed and the Projection is automatically restarted from the previously saved offset. This will result in that the call to the `ShoppingOrderService` will be retried. Since the Projection has `at-least-once` semantics the `ShoppingOrderService` must be idempotent, i.e. handle duplicate order attempts for the same `cartId`.
+If the call to the `ShoppingOrderService` fails the returned [.group-scala]#`Future[Done]`# [.group-java]#`CompletionStage<Done>`# is failed and the Projection is automatically restarted from the previously saved offset. This will result in that the call to the `ShoppingOrderService` will be retried. Since the Projection has `at-least-once` semantics the `ShoppingOrderService` must be idempotent, i.e. handle duplicate order attempts for the same `cartId`.
```
Comment (octonato) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-grpc-client.adoc:146]: fixed CompletionStage<Done>, leftover from previous PR (cc @ignasi35)

File: docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc [hunk 3]
```diff
@@ -163,11 +177,11 @@ Scala::
...
 ----
 
-<1> Kafka Consumer stream.
+<1> `RestartSource` will restart the stream in case of failures.
 
-<2> Offset is committed to Kafka when records have been processed.
+<2> Kafka Consumer stream.
 
-<3> `RestartSource` will restart the stream in case of failures.
+<3> Offset is committed to Kafka when records have been processed.
```
Comment (octonato) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:184]: swap ordering to follow appearance order in code

File: scripts/copy-identical-files.sh [hunk 4]
```diff
@@ -287,11 +287,6 @@ declare SRC="${tutorial_root}/shopping-cart-service-scala/src/main/resources/kaf
 cp ${SRC} ${tutorial_root}/05-shopping-cart-service-scala/src/main/resources/
 cp ${SRC} ${tutorial_root}/05-shopping-cart-service-java/src/main/resources/
 
-cp ${SRC} ${tutorial_root}/05-shopping-cart-service-java/src/main/resources/
-cp ${SRC} ${tutorial_root}/04-shopping-cart-service-java/src/main/resources/
-cp ${SRC} ${tutorial_root}/03-shopping-cart-service-java/src/main/resources/
-cp ${SRC} ${tutorial_root}/02-shopping-cart-service-java/src/main/resources/
-
```
Comment (octonato) [inline scripts/copy-identical-files.sh:294]: I don't know when this we introduced, but it's wrong. Previously we had the persistence.conf here. This is not moved upper in the file. Anyway, this should go as we were copying the kafka.conf everywhere

**Comment by octonato** [comment]
tests will fail for step 4, working on it

File: scripts/copy-identical-files.sh (hunk 4 above)
Comment (patriknw) [reply scripts/copy-identical-files.sh:294]: Must have been my mistake, when I changed the copy of persistence.conf

**Review by patriknw — APPROVED** [review]
LGTM

File: scripts/copy-identical-files.sh (hunk 4 above)
Comment (octonato) [reply scripts/copy-identical-files.sh:294]: I don't think so, I saw you remove it correctly. I can't explain how it came back.

**Comment by octonato** [comment]
Did another push here, I forgot to remove the publishing projection impl which is exactly what the step is about. 

We will need a thorough last pass when done. So many things that one (myself) can forget to remove. 🤦

**Comment by patriknw** [comment]
> We will need a thorough last pass when done. So many things that one (myself) can forget to remove.

I agree. I will not do thorough review of these PRs (or test them). I will read and test when we have all in place.

**Pull request merged**